services/datamanager/builtin/builtin.go

```diff
@@ -343,95 +340,59 @@ func (svc *builtIn) getCollectorFromConfig(attributes dataCaptureConfig) (data.C
-
-		// Sync existing files in captureDir.
-		var previouslyCaptured []string
-		//nolint
-		_ = filepath.Walk(svc.captureDir, func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return nil
-			}
-			if info.IsDir() {
-				return nil
-			}
-			previouslyCaptured = append(previouslyCaptured, path)
+		if info.IsDir() {
 			return nil
-		})
-		svc.syncer.Sync(previouslyCaptured)
-
-		// Validate svc.additionSyncPaths all exist, and create them if not. Then sync files in svc.additionalSyncPaths.
-		svc.syncer.Sync(svc.buildAdditionalSyncPaths())
-
-		// Kick off background routine to periodically sync files.
-		svc.startSyncBackgroundRoutine(intervalMins)
-	}
-	return nil
+		}
+		previouslyCaptured = append(previouslyCaptured, path)
+		return nil
+	})
+	svc.syncer.SyncCaptureFiles(previouslyCaptured)
 }
 
 // Sync performs a non-scheduled sync of the data in the capture directory.
-func (svc *builtIn) Sync(_ context.Context, extra map[string]interface{}) error {
+func (svc *builtIn) Sync(_ context.Context, _ map[string]interface{}) error {
+	svc.lock.Lock()
 	if svc.syncer == nil {
-		return errors.New("called Sync on data manager service with nil syncer")
+		err := svc.initSyncer(svc.cfg, 0.0)
+		if err != nil {
+			svc.lock.Unlock()
+			return err
+		}
 	}
-	err := svc.syncDataCaptureFiles()
-	if err != nil {
-		return err
+	if svc.syncDisabled {
+		svc.syncPreviouslyCaptured()
 	}
+	svc.lock.Unlock()
+	svc.syncDataCaptureFiles()
 	svc.syncAdditionalSyncPaths()
 	return nil
 }
 
-func (svc *builtIn) syncDataCaptureFiles() error {
+func (svc *builtIn) syncDataCaptureFiles() {
 	svc.lock.Lock()
-	oldFiles := make([]string, 0, len(svc.collectors))
-	currCollectors := make(map[componentMethodMetadata]collectorAndConfig)
-	for k, v := range svc.collectors {
-		currCollectors[k] = v
+	defer svc.lock.Unlock()
+	var queues []*datacapture.Queue
+	for _, collector := range svc.collectors {
+		queues = append(queues, collector.Collector.GetTarget())
 	}
-	svc.lock.Unlock()
-	for _, collector := range currCollectors {
-		// Create new target and set it.
-		attributes := collector.Attributes
-		captureMetadata, err := datacapture.BuildCaptureMetadata(attributes.Type, attributes.Name,
-			attributes.Model, attributes.Method, attributes.AdditionalParams, attributes.Tags)
-		if err != nil {
-			return err
-		}
+	svc.syncer.SyncCaptureQueues(queues)
 
-		nextTarget, err := datacapture.NewFile(svc.captureDir, captureMetadata)
-		if err != nil {
-			return err
-		}
-		oldFiles = append(oldFiles, collector.Collector.GetTarget().GetPath())
-		collector.Collector.SetTarget(nextTarget)
-	}
-	svc.syncer.Sync(oldFiles)
-	return nil
+	return
 }
 
+// TODO: pretty sure this will only be called in update. Ensure lock is held.
 func (svc *builtIn) buildAdditionalSyncPaths() []string {
-	svc.lock.Lock()
 	currAdditionalSyncPaths := svc.additionalSyncPaths
 	waitAfterLastModified := svc.waitAfterLastModifiedSecs
-	svc.lock.Unlock()
```

Comment:
[q]: [below](https://github.com/viamrobotics/rdk/blob/9c9b2d3fe069aba3510ee843fd58ae2513607eb6/services/datamanager/builtin/builtin.go#L401), why are we not handling the case when there's an error besides os.ErrNotExist